project/src/main/webapp/searchMessages.js

```diff
@@ -1,10 +1,11 @@
 function searchMessages() {
-    const query = getQuery();
-    if (query.length == 0) {
-        console.log("Please enter a question!");
+    const gmail = new GmailAPI();
```

Comment:
We probably also shouldn't be creating a new GmailAPI instance for every search, so can you also make this a global? I know I usually preach no globals, but in reality, you will need to have a few if you're not using a framework.

I think an ok pattern is to have these instances defined in like a main.js file and initialized in the window.onload function. Or create a main class to handle all the initialization.

```
class GesraMain {
  constructor() {
    this.gmail = new GmailAPI();
    this.gmailAuth = new GmailAuthentication();
  }
}

const gesra = new GesraMain();
```
```
function searchMessages() {
  gesra.gmail.getQuery();
  ...
}
```

Feel free to ignore for now and make these changes in a separate PR since it's a decent amount of refactoring.